mpisppy/utils/sputils.py

```diff
@@ -192,10 +192,12 @@ def create_EF(scenario_names, scenario_creator, scenario_creator_kwargs=None,
     # Check if every scenario has a specified probability
     probs_specified = \
         all([hasattr(scen, '_mpisppy_probability') for scen in scen_dict.values()])
-    if not probs_specified:
+    uniform_specified = \
+        probs_specified and all([scen._mpisppy_probability == "uniform" for scen in scen_dict.values()])
```

Comment:
- Since we changed this line -- we don't need to construct the intermediate list for `all`:
```suggestion
        probs_specified and all((scen._mpisppy_probability == "uniform") for scen in scen_dict.values())
```

Similar logic applied to line 194.

Reply:
good catch, good point, and thanks!